### DISCUSSION TITLE
What are the differences between useRoute of Vitepress rather than Vue?

### DISCUSSION THREAD

**Discussion by CodyBontecou** [vuejs/vitepress#1142] [Q&A] [answered]
I'm curious why we import `useRoute` from Vitepress rather than from Vue. Are there any changes occurring behind the scenes?

I'm running into an issue where the following Layout is not updating the `Post` component on route change.

```
<script setup>
import DefaultTheme from 'vitepress/theme'
import { useRoute } from 'vitepress'
import { ref, watch } from 'vue'

const { Layout } = DefaultTheme
const route = useRoute()
const currentRoute = ref(route)

watch(route, (newRoute, oldRoute) => {
  currentRoute.value = newRoute.path
})
</script>

<template>
  <Layout>
    <template #doc-after>
      <SimpleNewsletter />
      <Post
        repo="CodyBontecou/blog"
        theme="github-dark"
        :issue-term="currentRoute.value"
      />
    </template>
  </Layout>
</template>
```

A reproduction repo can be seen here - https://github.com/CodyBontecou/blog/tree/layout

**Comment by brc-dd** [comment, marked as answer]
I don't think there is any `useRoute` in `vue`. Regarding your issue, remove `.value` from template, it should work fine.

**Reply by CodyBontecou** [reply]
Yea, I'm starting to think it's the component itself that is creating this issue. I created an issue in the appropriate repo.

**Reply by CodyBontecou** [reply]
And you're right, `useRoute` is in vue-router, not vue.
